### DISCUSSION TITLE
ESP8266: How to forget/reset stored wifi credentials?

### DISCUSSION THREAD

**Discussion by dcski** [micropython/micropython#10857] [ESP8266]
My understanding is that the ESP8266 will internally save any successful wifi credentials and then automatically reconnected. How can I in my Micropython code delete/reset/remove all of those? Essentially building a "reset to factory settings function - just Micropython and my original code (.py), no user stored settings/data)

I came across this tutorial (https://hutscape.com/tutorials/erase-wifi-credentials) for a non-micropython ESP8266, but I can't figure out how to do it in Micropython - the functions aren't valid. 

Am I missing something very obvious or do I misunderstand how this works?

Thank you for any help,

**Comment by davefes** [comment]
Have a look in src and wifi.py
https://github.com/glenn20/micropython-espnow-utils
I use this with ESPNow on ESP32s so don't know if it will sort your particular problem. 

On second thought it looks like it will only guarantee that you start up in the correct state, ie STA versus AP

**Comment by glenn20** [comment]
@dcski  Micropython does not support this. In my wifi code I always execute a disconnect() after setting the STA_IF active and then force the connect in code.

I do have a [feature branch](https://github.com/micropython/micropython/compare/master...glenn20:micropython:esp8266-feature-reconnects) which adds an `autoconnect` option to WLAN.config() whic you can use to disable the autoconnect so-called feature of the esp8266. Once set, this persists across reboots until you change the setting. I haven't raised a PR for this as I'm not sure if there is much demand for it or support for inclusion in micropython. (In case it is not obvious - I really dislike this _"feature"_ of the esp8266).

**Reply by glenn20** [reply]
Whoops - I misremembered. I did make a PR: #9460.
